File: AppMobile/8180378/app/src/main/java/com/example/a8180378/models/Absence.java
package com.example.a8180378.models;

/**
 * The Absence class stores information about the Absence
 */
public class Absence {

    /**
     * contains the date of the absence
     */
    private String date;

    /**
     * contains the description of the absence
     */
    private String description;

    /**
     * contains the description of the absence
     */
    private AbsenceType absenceType;

    /**
     * contains the driver of the absence
     */
    private String id;

    /**
     * Getters e Setters
     */

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public AbsenceType getAbsenceType() {
        return absenceType;
    }

    public void setAbsenceType(AbsenceType absenceType) {
        this.absenceType = absenceType;
    }
}

File: BackEnd/WebApi/WebApi/models/Absence.cs
using System.ComponentModel.DataAnnotations;
using WebApplication1.Global.Enumerations;

namespace WebApplication1.Models;

/// <summary>
///  The Absence class stores information about the Absence
/// </summary>
public class Absence
{
    /// <summary>
    /// identifier of the Absence
    /// </summary>
    public Guid id { get; set; }

    /// <summary>
    /// contains the driver who registered the absence
    /// </summary>
    [Required]
    public virtual User driver { get; set; }

    /// <summary>
    /// contains the date and time of the absence
    /// </summary>
    public DateTime date { get; set; }

    /// <summary>
    /// contains the description of the absence
    /// </summary>
    public string description { get; set; }

    /// <summary>
    /// contains the type of the absence
    /// </summary> 
    public AbsenceType absence { get; set; }

    /// <summary>
    /// contains the state of the absence
    /// </summary> 
    public Status status { get; set; }
}